File: JavaScript/loginregister.js
"use strict";

// Data Storage of Accounts
const accounts = [
  {
    firstName: "Ley",
    lastName: "Jivs",
    age: 22,
    username: "ley",
    password: "ley12345",
    email: "john@gmail.com",
  },
  {
    firstName: "Zero",
    lastName: "Johnson",
    age: 34,
    username: "zero",
    password: "zero12345",
    email: "zero@gmail.com",
  },

  {
    firstName: "Luxus",
    lastName: "Smith",
    age: 27,
    username: "luxus",
    password: "luxus12345",
    email: "luxus@gmail.com",
  },
];

// Display Accounts Data
const displayAccountsData = () => {
  console.log("----- Data Accounts Details ------");
  accounts
    .slice()
    .sort((a, b) => a.firstName.localeCompare(b.firstName))
    .map((acc, i) => {
      console.log(
        `${i + 1}: First name: ${acc.firstName} lastname: ${
          acc.lastName
        } age: ${acc.age} username: ${acc.username} password: ${
          acc.password
        } email: ${acc.email}`
      );
    });
};

// Function for Reusable Asking User for Registration
const askUserRegistration = () => {
  const askName = prompt("Enter name: ");
  const askLastName = prompt("Enter last name: ");
  const askAge = Number(prompt("Enter age:"));
  const askUserName = prompt("Enter username: ");

  let askPassword;
  while (true) {
    askPassword = prompt("Enter password: ");
    if (/\d/.test(askPassword)) {
      break;
    } else {
      alert("Invalid password! should have number");
    }
  }

  let askEmail;
  // Keep asking for a valid email address
  while (true) {
    askEmail = prompt("Enter email: ");
    if (askEmail.includes("@")) {
      accounts.push({
        firstName: askName,
        lastName: askLastName,
        age: askAge,
        username: askUserName,
        password: askPassword,
        email: askEmail,
      });
      alert("Registration successful!");
      askLoginOrRegister(); // Prompt user to log in or register again
      return; // Exit the function after registration
    } else {
      alert("Invalid email! should have @");
    }
  }
};

// Function for Reusable Asking User for Login
const askUserLogin = () => {
  const askUserName = prompt("Enter username: ");
  const askPassword = prompt("Enter password: ");

  const foundAccount = accounts.find(
    account =>
      account.username === askUserName && account.password === askPassword
  );

  if (foundAccount) {
    displayAccountsData();
  } else {
    alert("Invalid account!");
    askLoginOrRegister(); // Prompt user to log in or register again
  }
};

// Function for Reusable Asking User for Login or Register
const askLoginOrRegister = () => {
  const askUser = prompt("Login or Register?");

  if (askUser === "login") {
    askUserLogin();
  } else if (askUser === "register") {
    askUserRegistration();
  } else {
    alert("Invalid input! should be Login or Register.");
    askLoginOrRegister();
  }
};

// First Step Ask User to Login or Register
askLoginOrRegister();

File: Python/loginregister.py
# Data Storage of Accounts
accounts = [
    {
        "firstName": "Ley",
        "lastName": "Jivs",
        "age": 22,
        "username": "ley",
        "password": "ley12345",
        "email": "ley@gmail.com",
    },
    {
        "firstName": "Zero",
        "lastName": "Johnson",
        "age": 34,
        "username": "zero",
        "password": "zero12345",
        "email": "zero@gmail.com",
    },
    {
        "firstName": "Luxus",
        "lastName": "Smith",
        "age": 27,
        "username": "luxus",
        "password": "luxus12345",
        "email": "luxus@gmail.com",
    },
]

# Function to sort and display accounts
def display_sorted_accounts():
    print("----- Data Accounts Details ------")

    # Sort all accounts by "firstName" (case-insensitive)
    sorted_accounts = sorted(accounts, key=lambda acc: acc["firstName"].lower())

    # Iterate through the sorted accounts and print the details
    for i, acc in enumerate(sorted_accounts):
        print(f"{i + 1}: First name: {acc['firstName']} lastname: {acc['lastName']} age: {acc['age']} username: {acc['username']} password: {acc['password']} email: {acc['email']}")

# Function for Reusable Asking User for Registration
def ask_user_registration():
    while True:
        ask_name = input("Enter name: ")
        ask_last_name = input("Enter last name: ")
        ask_age = int(input("Enter age: "))
        ask_username = input("Enter username: ")

        while True:
            ask_password = input("Enter password: ")
            if any(char.isdigit() for char in ask_password):
                break
            else:
                print("Invalid password! Should have a number")

        while True:
            ask_email = input("Enter email: ")
            if "@" in ask_email:
                new_account = {
                    "firstName": ask_name,
                    "lastName": ask_last_name,
                    "age": ask_age,
                    "username": ask_username,
                    "password": ask_password,
                    "email": ask_email,
                }
                accounts.append(new_account)
                print("Registration successful!")

                ask_login_or_register()  # Call the login or register function again

            else:
                print("Invalid email! Should have '@'")

# Function for Reusable Asking User for Login
def ask_user_login():
    ask_username = input("Enter username: ")
    ask_password = input("Enter password: ")

    found_account = next((account for account in accounts if account["username"] == ask_username and account["password"] == ask_password), None)

    if found_account:
        print("Login successful!")
        display_sorted_accounts()  # Display sorted accounts after login
        exit()  # End the program
    else:
        print("Invalid account!")
        ask_login_or_register()  # Prompt user to log in or register again

# Function for Reusable Asking User for Login or Register
def ask_login_or_register():
    ask_user = input("Login or Register?")

    if ask_user == "login":
        ask_user_login()
    elif ask_user == "register":
        ask_user_registration()
    else:
        print("Invalid input! Should be 'Login' or 'Register'.")
        ask_login_or_register()

# First Step: Ask User to Login or Register
ask_login_or_register()